pennylane/devices/qubit/simulate.py

```diff
@@ -456,20 +446,19 @@ def parse_native_mid_circuit_measurements(
     Returns:
         tuple(TensorLike): The results of the simulation
     """
-    idx_one_shot = find_not_measurement_values(circuit)
-    normalized_meas = [None] * len(circuit.measurements)
-    for i, m in zip(idx_one_shot, all_shot_meas):
-        normalized_meas[i] = gather_non_mcm(circuit.measurements[i], m, mcm_shot_meas)
-
     idx_sample = find_measurement_values(circuit)
-    counter = Counter()
-    if any(isinstance(m, (CountsMP, ProbabilityMP)) for m in circuit.measurements):
-        for d in mcm_shot_meas:
-            counter.update(d)
-    for i in idx_sample:
-        normalized_meas[i] = gather_mcm(
-            circuit.measurements[i], circuit.measurements[i].mv, mcm_shot_meas, counter
-        )
+    normalized_meas = [None] * len(circuit.measurements)
```

Comment:
I'm just curious about the difference between preallocating the array and then updating it versus initializing it to `[]` and appending to it.  If we are appending to a list instead of updating a list, we could get rid of the dependence on `i`.

Reply:
Let's do it like that, looks more pythonic. We still need the dependence on `i` because we may need to pass `all_shot_meas[i]`. We can't zip it with `circuit.measurements` since it may be `None` in certain cases.